x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/completion/AzureAiStudioChatCompletionRequestTaskSettings.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.azureaistudio.completion;
+
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.inference.ModelConfigurations;
+
+import java.util.Map;
+
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalBoolean;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalDouble;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalPositiveInteger;
+import static org.elasticsearch.xpack.inference.services.azureaistudio.AzureAiStudioConstants.DO_SAMPLE_FIELD;
+import static org.elasticsearch.xpack.inference.services.azureaistudio.AzureAiStudioConstants.MAX_NEW_TOKENS_FIELD;
+import static org.elasticsearch.xpack.inference.services.azureaistudio.AzureAiStudioConstants.TEMPERATURE_FIELD;
+import static org.elasticsearch.xpack.inference.services.azureaistudio.AzureAiStudioConstants.TOP_P_FIELD;
+
+public record AzureAiStudioChatCompletionRequestTaskSettings(
+    @Nullable Double temperature,
+    @Nullable Double topP,
+    @Nullable Boolean doSample,
+    @Nullable Integer maxNewTokens
+) {
+
+    public static final AzureAiStudioChatCompletionRequestTaskSettings EMPTY_SETTINGS = new AzureAiStudioChatCompletionRequestTaskSettings(
+        null,
+        null,
+        null,
+        null
+    );
+
+    /**
+     * Extracts the task settings from a map. All settings are considered optional and the absence of a setting
+     * does not throw an error.
+     *
+     * @param map the settings received from a request
+     * @return a {@link AzureAiStudioChatCompletionRequestTaskSettings}
+     */
+    public static AzureAiStudioChatCompletionRequestTaskSettings fromMap(Map<String, Object> map) {
+        if (map.isEmpty()) {
+            return AzureAiStudioChatCompletionRequestTaskSettings.EMPTY_SETTINGS;
+        }
+
+        ValidationException validationException = new ValidationException();
+
+        Double temperature = extractOptionalDouble(map, TEMPERATURE_FIELD);
```

Comment:
> Should we validate that they're in the correct range? 

Good question... I'd say yes - _but_ there's no direct docs I can see for the valid ranges for any of these parameters... the only thing that comes close is in the AzureOpenAI .DLL / SDK documentation:
* [Temperature](https://learn.microsoft.com/en-us/dotnet/api/azure.ai.openai.chatcompletionsoptions.temperature?view=azure-dotnet-preview) : valid range of 0.0 to 2.0
* [Top P aka Nuclear Sampling](https://learn.microsoft.com/en-us/dotnet/api/azure.ai.openai.chatcompletionsoptions.nucleussamplingfactor?view=azure-dotnet-preview) - no range specified, but I'd assume it's the same as temperature
* [Max Tokens](https://learn.microsoft.com/en-us/dotnet/api/azure.ai.openai.chatcompletionsoptions.maxtokens?view=azure-dotnet-preview) minimum of 0

> I wonder if we're getting an error response back but not passing it along.

This I doubt as we're still getting a 200 response - however, I can certainly see if all the probabilities are negative it might only consider those > 0.0... but 🤷 I don't know for certain and will do a bit of testing manually...

Reply:
> > I wonder if we're getting an error response back but not passing it along.
> 
> This I doubt as we're still getting a 200 response - however, I can certainly see if all the probabilities are negative it might only consider those > 0.0... but 🤷 I don't know for certain and will do a bit of testing manually...

FYI - I ran a manual test - and yep - no error. Calling with `temperature` of -2.0 directly to the `.../score` endpoint yields a 200 response with the following:
```
{
    "output": "None"
}
```